lib/Parse/Lexer.cpp

```diff
@@ -656,6 +663,13 @@ static bool isRightBound(const char *tokEnd, bool isLeftBound,
     // "^.y" to be a prefix op, not binary.
     return !isLeftBound;
 
+  case '/':
+    // The start of a comment counts as whitespace, so it's not right bound.
+    if (tokEnd[1] == '/' || tokEnd[1] == '*')
+      return false;
+    else
+      return true;
+
```

Comment:
I don't understand what this means.  Comments aren't returned as tokens, so the notion of whether the comment is right bound doesn't seem relevant.  I think this code is dead.